orion/handlers/http/defaults.go

```diff
@@ -0,0 +1,66 @@
+package http
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/carousell/Orion/orion/modifiers"
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/gogo/protobuf/proto"
+	"github.com/gorilla/mux"
+	"github.com/gorilla/websocket"
+	"github.com/mitchellh/mapstructure"
+)
+
+// DefaultEncoder encodes a HTTP request if none are registered. This encoder
+// populates the proto message with URL route variables or fields from a JSON
+// body if either are available.
+func DefaultEncoder(req *http.Request, r interface{}) error {
+	// check and map url params to request
+	params := mux.Vars(req)
+	if len(params) > 0 {
+		mapstructure.Decode(params, r)
+	}
+
+	data, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		if err == io.EOF {
+			return nil
+		}
+		return err
+	}
+
+	return deserialize(req.Context(), data, r)
+}
+
+func deserialize(ctx context.Context, data []byte, r interface{}) error {
+	serType := ContentTypeFromHeaders(ctx)
+	switch serType {
+	case modifiers.ProtoBuf:
+		if protoReq, ok := r.(proto.Message); ok {
+			return proto.Unmarshal(data, protoReq)
+		}
+		fallthrough
```

Comment:
bit unsafe, if content type is for proto then if there isnt a proto message in the request, should fail?

Reply:
yes the function should fail and return the error, but in case of get requests we check for io.EOF and don't call this function